resources.md

```diff
@@ -32,8 +32,27 @@ It's often said that the best way to learn to write programs is to write program
 
 - [Project Euler](https://projecteuler.net/) is great fun for the mathematically-minded. Over 650 problems which ramp up rather quickly in difficulty. If you've got a strong mathematical background you'll have a real advantage here. If not, you might well acquire one.
 
-## Resources that cost money
+## Online resources that cost money
 
 - [Python Morsels](https://www.pythonmorsels.com/): A short Python exercise every week, followed a fews days later by a number of idiomatic solutions, with explanations of each one.
 
 - [PyBites](https://codechalleng.es/): <q>Hone your Python skills by coding and verifying exercises in the comfort of your browser.</q> Offers a small number of free exercises, but most require buying a bundle or monthly subscription. Exercises are grouped by skill level (Bites of Py) and topic (Learning Paths).
+
+## Books
+### There are thousands of books on programming in python. This page will focus on titles helpful for beginners and advanced beginners. Most of the titles listed are available either free online or through a Boston-area library system ([BPL](bpl.org) or [Minuteman](minlib.net)), as well as for purchase through your favorite bookseller. 
+### Please note that this page was originally compiled by a non-beginner. If you are a beginner, we will be grateful for your feedback on these suggestions. 
+
+- [Think Python!](https://greenteapress.com/wp/think-python-2e/) (Allen Downey). Currently our favorite title for beginners, Downey starts from the most basic programming concepts and develops a computer-science mindset as well as python fluency. While no book is perfect for all readers, this one has found broad appeal with many beginners. We recommend that you start here. Free PDF and HTML versions available at https://greenteapress.com/wp/think-python-2e/, and paper copies are available from booksellers. (O'Reilly/Green Tea Press) 
+
+- [Learn Python the Hard Way](https://learnpythonthehardway.org/python3/) (Zed Shaw). Another popular title for beginners. Shaw's book is structured as a series of exercises, with a strong focus on hands-on learning. Some beginners have reported finding his highly opinionated style to be off-putting. Sample chapters available on the book's [website](https://learnpythonthehardway.org/python3/) are representative of the book's style. Digital edition (not free) available at https://learnpythonthehardway.org/python3/ and paper editions widely available. (pub???) (minlib, bpl)
```

Comment:
Given "his highly opinionated style" (which I've encountered elsewhere), I'm reluctant to include this.

Reply:
That's fair. My argument for including it would be: it's a popular title, and people will probably get it as a recommendation, therefore it's worth letting people know that there are reservations about. 
If this argument is not convincing, feel free to remove it in your edits.